docs/cadl-for-openapi-dev.md

```diff
@@ -0,0 +1,285 @@
+| `404`            | `NotFoundResponse`     |                                      |
+| `409`            | `ConflictResponse`     |                                      |
+
+If the first return type does not contain a `statusCode` header, it is assumed to be the `200` response.
+Any return type after the first that does not contain a `statusCode` header is assumed to be the `default` response.
+
+### Response Object
+
+In OpenAPI, a response object [[v2][v2-response], [v3][v3-response]] describes a single response for an operation.
+The structure of the response object changed significantly from OpenAPI v2 to OpenAPI v3, but there are many
+elements common to both.
+
+[v2-response]: https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md#responseObject
+[v3-response]: https://github.com/OAI/OpenAPI-Specification/blob/3.0.3/versions/3.0.3.md#responseObject
+
+The fields in an OpenAPI response object are specified with the following Cadl constructs:
+
+| OpenAPI `response` field | Cadl construct                                     | Notes                    |
+| ------------------------ | -------------------------------------------------- | ------------------------ |
+| `description`            | `@doc` decorator                                   |                          |
+| `schema`                 | return type                                        |                          |
+| `headers`                | fields in the return type with `@header` decorator |                          |
+| `examples`               |                                                    | Not currently supported. |
+| `links` (OAS3)           |                                                    | Not currently supported. |
+
+## Schema Object
+
+OpenAPI schemas are represented in Cadl by [models](https://github.com/microsoft/cadl/blob/main/docs/tutorial.md#models).
+Models have any number of members and can extend and be composed with other models.
+
+Models can be defined with the `model` statement and then referenced by name, which generally results in a `$ref` to a schema for the model in the `definitions` or `components.schemas` section of the OpenAPI document.
+
+Cadl supports the ["spread" operator](https://github.com/microsoft/cadl/blob/main/docs/tutorial.md#spread) (`...`), which copies the members of the source model into the target model.
+But Cadl processes all spread transformations before emitters are invoked, so this form of reuse is not represented in the emitted OpenAPI.
+
+Cadl also supports single inheritance of models with the `extends` keyword. This construct can be used to produce an `allOf` with a single element (the parent schema) in OpenAPI. Cadl does not current provide a means to produce an `allOf` with more than one element -- these are generally treated as "composition" in code generators and thus better represented in Cadl with the spread operator.
```

Comment:
Can we have an example of `extends` and `....` so they know where to use it without having to look at the main tutorial